### DISCUSSION TITLE
How to mock window.matchMedia - Test environment Jest throws "TypeError: window.matchMedia is not a function"

### DISCUSSION THREAD

**Discussion by Wezea** [tinymce/tinymce-react#358] [Q&A]
Hi there :-)

tinynce-react is great and doing its thing. In the browser we do not have any issues. However, running our tests causes all test suites to fail. Jest is throwing this exception:

`TypeError: window.matchMedia is not a function`

We have followed the [Jest manual](https://jestjs.io/docs/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom)  and added a mock for `window.matchMedia` but still this is not working.

How could we get access to the window object within the TinyMCE instance and mock `window.matchMedia`?

What I can't wrap my head around: Even tests without any TinyMCE dependency are crashing with this exception.

We are running: 

`tinymce: 5.10.5`
`@tinymce/tinymce-react`: `3.14.0`

We would highly appreciate some help, as all our tests are currently failing.

**Comment by a-wf** [comment]
any solution?
